### PR TITLE
[MI-2400] Integrate request modal with the API

### PR DESCRIPTION
**Pull request by ayusht2810**
image

### CODE REVIEW

File: webapp/src/services/index.ts [hunk 1]
```diff
@@ -108,6 +108,14 @@ const pluginApi = createApi({
                 body,
             }),
         }),
+        [Constants.pluginApiServiceConfigs.searchItems.apiServiceName]: builder.query<RequestData[], SearchItemsParams>({
+            query: ({search, perPage}) => ({
+                headers: {[Constants.HeaderCSRFToken]: Cookies.get(Constants.MMCSRF)},
+                url: Constants.pluginApiServiceConfigs.searchItems.path,
+                method: Constants.pluginApiServiceConfigs.searchItems.method,
+                params: {search, perPage: perPage || 10},
```
Comment (Kshitij-Katiyar) [inline webapp/src/services/index.ts:116]: please create a constant for perPage default value like here it is 10

**Review by Kshitij-Katiyar — APPROVED** [review]

File: webapp/src/containers/createRequest/index.tsx [hunk 2]
```diff
@@ -123,8 +90,51 @@ const CreateRequest = () => {
...
+    const getItemsSuggestions = () => {
+        const {isLoading, isSuccess, isError, data, error: apiErr} = getApiState(Constants.pluginApiServiceConfigs.searchItems.apiServiceName, searchItemsPayload);
+        return {isLoading, isSuccess, isError, data: data as RequestData[], error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
+    };
+
+    // Get the suggestions from the API
+    const getSuggestions = ({searchFor}: {searchFor?: string}) => {
+        setApiError(null);
+        setRequest(null);
+        setSearchItemsPayload({search: searchFor || ''});
```
Comment (manojmalik20) [inline webapp/src/containers/createRequest/index.tsx:102]: If `searchFor` is `''`, then I don't think we should make the API call.

File: webapp/src/containers/createRequest/index.tsx [hunk 3]
```diff
@@ -123,8 +90,51 @@ const CreateRequest = () => {
...
+        makeApiRequest(Constants.pluginApiServiceConfigs.searchItems.apiServiceName, {search: searchFor || ''});
+    };
 
+    const debouncedGetSuggestions = useCallback(Utils.debounce(getSuggestions, 500), [getSuggestions]);
```
Comment (manojmalik20) [inline webapp/src/containers/createRequest/index.tsx:106]: There should be a constant for `500` and it should be used in all the search fields.

File: webapp/src/containers/createRequest/index.tsx [hunk 4]
```diff
@@ -137,58 +147,71 @@ const CreateRequest = () => {
...
+                                                <span className='search-panel__description-header margin-right-10 text-ellipsis'>{header.label}</span>
+                                                <span className='search-panel__description-text channel-text wt-500 text-ellipsis white-space-inherit'>{showModalLoader ? <SkeletonLoader/> : request[header.key] || 'N/A'}</span>
+                                            </li>
+                                        ))}
+                                    </ul>
+                                    {serviceNowBaseURL && (
+                                        <div>
+                                            <a
+                                                className='color--link btn btn-primary servicenow-request-button'
+                                                href={`${serviceNowBaseURL}/${Constants.REQUEST_BASE_URL}${request.id}`}
```
Comment (manojmalik20) [inline webapp/src/containers/createRequest/index.tsx:194]:
```suggestion
                                                href={`${serviceNowBaseURL}/${Constants.REQUEST_BASE_URL}/${request.id}`}
```
Remove `/` from the end of the constant `REQUEST_BASE_URL`.

Comment (ayusht2810) [reply webapp/src/containers/createRequest/index.tsx:196]: there is no `/` at end of URL, we are using id as a param

Comment (ayusht2810) [reply webapp/src/containers/createRequest/index.tsx:196]: there is no `/` at end of URL, we are using id as a param

**Pull request merged**